packages/account/src/providers/provider.ts

```diff
@@ -416,11 +416,16 @@ export default class Provider {
   cache?: ResourceCache;
 
   /** @hidden */
-  static clearChainAndNodeCaches() {
+  private static clearChainAndNodeCaches() {
     Provider.nodeInfoCache = {};
     Provider.chainInfoCache = {};
   }
 
+  /** @hidden */
+  clearChainAndNodeCaches() {
+    Provider.clearChainAndNodeCaches();
+  }
+
```

Comment:
Because `clearChainAndNodeCaches` is static, calling `Provider.clearChainAndNodeCaches()` was not clearing the cache for already instantiated providers:

```ts
const provider1 = await Provider.create(URL);

// Attempt to clear the chain and node information cache.
Provider.clearChainAndNodeCaches();

/**
 * Fetching the chain and node information using `fetchChainAndNodeInfo`
 * on the existing instance (`provider1`) won't reflect the cache being cleared.
 * The cache was not cleared and is still used.
 */
const persistentCachedData = await provider1.fetchChainAndNodeInfo();

/**
 * When creating a new instance with the static method `.create`,
 * it refetches the chain and node information. In this scenario, the cache
 * is properly cleared and refreshed, likely due to `.create` being a static method.
 */
const provider2 = await Provider.create(URL);
```

Therefore I've made `Provider.clearChainAndNodeCaches` private and used it within `clearChainAndNodeCaches`

Reply:
What do you mean by "clearing the cache for instantiated providers"?

If we clear the cache and then call `getChain()`, it should re-fetch.

https://github.com/FuelLabs/fuels-ts/blob/ead7761e94f9c2cfc19353a31a93a9ad58c14f07/packages/account/src/providers/provider.ts#L555

Reply:
> If we clear the cache and then call getChain(), it should re-fetch.

We'd have to make the `getChain` method async for that - something we talked about when the caching of chain info was introduced but I can't find the discussion on GitHub.

Reply:
Exactly. That is the direction that I wanted go. But will add more breaking changes